src/main/java/com/amazon/opendistroforelasticsearch/security/support/wildcard/Wildcard.java

```diff
@@ -0,0 +1,70 @@
+package com.amazon.opendistroforelasticsearch.security.support.wildcard;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+// Case-sensitivity and other options are defined by concrete Wildcard implementation
+public interface Wildcard {
+    boolean matches(String candidate);
+
+    default boolean matchesAny(Collection<String> candidates) {
+        return candidates.stream().anyMatch(this::matches);
+    }
+
+    default boolean matchesAny(String[] candidates) {
+        return Arrays.stream(candidates).anyMatch(this::matches);
+    }
+
+    // TODO: make serializable, hashable etc.
+    Wildcard ANY = new Wildcard() {
```

Comment:
Needs to implement `matchAny` and assert if `matches` (`test`) is called.

Reply:
Care to elaborate about matches(test) ?

Reply:
Please ignore my comment regarding `assert`, it may still be a valid call.